Framework/jmaqs-selenium/src/main/java/com/magenic/jmaqs/selenium/WebDriverFactory.java

```diff
@@ -0,0 +1,554 @@
+   * @param remoteBrowserVersion The remote browser version
+   * @param remoteCapabilities   Additional remote capabilities
+   * @return The remote driver options
+   */
+  public static MutableCapabilities getRemoteOptions(RemoteBrowserType remoteBrowser, String remotePlatform,
+      String remoteBrowserVersion, Map<String, Object> remoteCapabilities) {
+    MutableCapabilities options = null;
+    switch (remoteBrowser) {
+      case IE:
+        options = new InternetExplorerOptions();
+        break;
+
+      case Firefox:
+        options = new FirefoxOptions();
+        break;
+
+      case Chrome:
+        options = new ChromeOptions();
+        break;
+
+      case Edge:
+        options = new EdgeOptions();
+        break;
+
+      case Safari:
+        options = new SafariOptions();
+        break;
+
+      default:
+        throw new IllegalArgumentException(
+            StringProcessor.safeFormatter("Remote browser type '%s' is not supported", remoteBrowser));
+    }
+
+    // Make sure the remote capabilities dictionary exists
+    if (remoteCapabilities == null) {
+      remoteCapabilities = new HashMap<>();
+    }
+
+    // Add a platform setting if one was provided
+    if (!remotePlatform.isEmpty() && !remoteCapabilities.containsKey("platform")) {
+      remoteCapabilities.put("platform", remotePlatform);
+    }
+
+    // Add a remote browser setting if one was provided
+    if (!remoteBrowserVersion.isEmpty() && !remoteCapabilities.containsKey("version")) {
+      remoteCapabilities.put("version", remoteBrowserVersion);
+    }
+
+    // Add additional capabilities to the driver options
+    setDriverOptions(options, remoteCapabilities);
+
+    return options;
+  }
+
+  /**
+   * Add additional capabilities to the driver options.
+   *
+   * @param driverOptions          The driver option you want to add capabilities to
+   * @param additionalCapabilities Capabilities to add
+   */
+  public static void setDriverOptions(MutableCapabilities driverOptions, Map<String, Object> additionalCapabilities) {
+    // If there are no additional capabilities just return
+    if (additionalCapabilities == null) {
+      return;
+    }
+
+    additionalCapabilities.forEach((key, value) -> {
+      if ((value instanceof String) && !((String) value).isEmpty()) {
+        driverOptions.setCapability(key, value);
+      }
+    });
+  }
+
+  /**
+   * Sets the browser size based on the provide string value.
+   * Browser size is expected to be: MAXIMIZE, DEFAULT, or #x# (such as 1920x1080).
+   * MAXIMIZE just maximizes the browser.
+   * DEFAULT does not change the current size.
+   * #x# sets a custom size.
+   *
+   * @param webDriver the webDriver from the Browser method
+   * @param size      Browser size in the following format: MAXIMIZE, DEFAULT, or #x# (such as 1920x1080)
+   */
+  public static void setBrowserSize(WebDriver webDriver, String size) {
+    size = size.toUpperCase();
+
+    if (size.equals(MAXIMIZE)) {
```

Comment:
Since this is a variable and the string is the same I recommend just having a shorter var name like MAX or use a longer more detailed name like WINDOW_MAX